packages/test/src/plugin.ts

```diff
@@ -33,24 +33,32 @@ function isBasicObject(o: unknown): o is object {
   return proto === Object.prototype || proto === null
 }
 
-function withoutUndefinedProperties(o: object): object {
-  const copy = Object.create(
-    Object.getPrototypeOf(o) as object | null,
-    Object.getOwnPropertyDescriptors(o),
-  ) as object
-
-  for (const key of Reflect.ownKeys(copy)) {
-    const descriptor = Reflect.getOwnPropertyDescriptor(copy, key)!
+function withoutUndefinedProperties(
+  source: object,
+  deep: boolean,
+  copy = true,
+): object {
+  const object = copy
+    ? (Object.create(
+        Object.getPrototypeOf(source) as object | null,
+        Object.getOwnPropertyDescriptors(source),
+      ) as object)
+    : source
+
+  for (const key of Reflect.ownKeys(object)) {
+    const descriptor = Reflect.getOwnPropertyDescriptor(object, key)!
     if (
       descriptor.enumerable &&
       descriptor.configurable &&
       descriptor.value === undefined
     ) {
-      Reflect.deleteProperty(copy, key)
+      Reflect.deleteProperty(object, key)
+    } else if (deep && isBasicObject(descriptor.value)) {
+      withoutUndefinedProperties(descriptor.value, true, false)
     }
   }
 
-  return copy
+  return object
 }
```

Comment:
This is pretty nasty code because it needs to be to deal with JS weirdness but essentially I just changed it to be recursive if desired.